R/CreatePackageVignette.R

```diff
@@ -0,0 +1,101 @@
+#' @title pkgnet Report as Vignette
+#' @name CreatePackageVignette
+#' @description Create pkgnet package report as an Rmarkdown vignette. This
+#'    vignette is able to be rendered with the
+#'    \code{\link[knitr:vignette_engines]{knitr::rmarkdown}} vignette engine
+#'    into HTML vignettes upon package building. It is also compatible with
+#'    \code{\link[pkgdown:build_articles]{pkgdown}} sites.
+#' @param pkg_name (string) name of a package
+#' @param pkg_path (string) The path to the package repository. If given, coverage
+#'                 will be calculated for each function. \code{pkg_path} can be an
+#'                 absolute or relative path.
+#' @param pkg_reporters (list) a list of initialized package reporters
+#' @param vignette_path (string) The path and filename of the output vignette
+#'    file. The default assumes your working directory is the package root.
+#' @importFrom rlang enexpr
+#' @importFrom assertthat assert_that is.string is.readable
+#' @importFrom tools file_ext
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
+                                  , pkg_path = NULL
+                                  , pkg_reporters = list(
+                                      DependencyReporter$new()
+                                      , FunctionReporter$new()
+                                  )
+                                  , vignette_path = file.path("vignettes"
```

Comment:
@jayqi As it is currently written, it is very easy to create a vignette for a different package within your package.  Since you are automatically deriving the path here to the vignette, could we automatically cross reference the given package name and the folder path package name?  If they do not match, just return a warning message at the end of the successful run.

Reply:
Your directories don't have to be the same as the package name. Like [uptasticsearch's package root is called `r-pkg`](https://github.com/UptakeOpenSource/uptasticsearch/tree/master/r-pkg).

Reply:
True, but this is unconventional right?  It may still be worth a warning.  A vignette for another package in your package seems like a serious error.

Reply:
@bburns632 I'm fine with the warning not error (as we discussed IRL) but FTR it isn't THAT unconventional. Here are some high-visibility projects that don't follow that convention:

* [`LightGBM`](https://github.com/Microsoft/LightGBM): `LightGBM/R-package/` (note the package is `lightgbm` all lowercase)
* [`feather`](https://github.com/wesm/feather/): `feather/R/`
* [`XGBoost`](https://github.com/dmlc/xgboost): `xgboost/R-package/`
* [`CatBoost`](https://github.com/catboost/catboost/tree/master/catboost): `catboost/catboost/R-package/`

Reply:
My latest change has warnings for (1) if you write to a directory named `vignettes` that is not in a package root or (2) if you write to a directory named `vignettes` in a package root that is for a different package than `pkg_name`.

Reply:
👍